Workspaces/Base/WorkspaceUI.cs

```diff
@@ -12,7 +13,7 @@
 
 namespace UnityEditor.Experimental.EditorVR.Workspaces
 {
-	sealed class WorkspaceUI : MonoBehaviour, IUsesStencilRef, IUsesViewerScale, IGetPointerLength
+	sealed class WorkspaceUI : MonoBehaviour, IUsesStencilRef, IUsesViewerScale, IGetPointerLength, IControlHaptics, IRayToNode, IUsesNode
```

Comment:
Haptics shouldn't be driven by the UI; Bring up to the level of Workspace and create events here if necessary.